portfolio/src/main/webapp/script.js

```diff
@@ -61,28 +61,60 @@ function filterCourseDisplay(courseType = 'course') {
 }
 
 /**
- * Fetch data from server and display on DOM.
+ * Fetch comments from server and display on DOM.
  */
-function getAndDisplayComments() {
+function loadComments() {
   fetch('/list-comments')
   .then(response => response.json())
   .then((data) => {
-    // Display each comment in a block with a like button .
+    const display = document.getElementById('content');
     for (let comment of data) {
-      const commentContainer = document.createElement('div');
-      commentContainer.class = 'comment-container';
-      commentContainer.innerHTML = '';
-      commentContainer.innerHTML += 'Posted by: ' + comment.name + '<br>';
-      commentContainer.innerHTML += comment.content + '<br>';
-      commentContainer.innerHTML += comment.numLikes + ' likes<br>';
-      commentContainer.innerHTML += 'Posted at ' + comment.timestamp + '<br>';
-      // TODO: use methods to create button.
-      const likeButtonHTML = '<form action="/add-like" method="POST">' + 
-        '<input type="hidden" name="comment-key" value="' + comment.key + '" />' + 
-        '<input type="submit" value="Like" class="btn btn-danger btn-sm" /></form>';
-      commentContainer.innerHTML += likeButtonHTML;
-      commentContainer.innerHTML += '<br><br>';
-      document.getElementById('content').appendChild(commentContainer);
+      display.appendChild(createCommentElement(comment));
     }
   });
-}
\ No newline at end of file
+}
+
+/**
+ * Create HTML display for one comment.
+ */
+function createCommentElement(comment) {
+  const commentContainer = document.createElement('div');
+
+  const commentText = document.createElement('p');
+  commentText.innerHTML += 'Posted by: ' + comment.name + '<br>';
+  commentText.innerHTML += comment.content + '<br>';
+  commentText.innerHTML += comment.numLikes + ' likes<br>';
+  commentText.innerHTML += 'Posted at ' + comment.timestamp;
+
+  const likeButton = createLikeButton(comment.key);
+
+  commentContainer.appendChild(commentText);
+  commentContainer.appendChild(likeButton);
+  commentContainer.innerHTML += '<br><br>';
+
+  return commentContainer;
+}
+
+/**
+ * Create a like button for a comment given the comment's unique key.
+ */
+function createLikeButton(key) {
+  const likeForm = document.createElement('form');
+  likeForm.action = '/add-like';
+  likeForm.method = 'POST';
+
+  const keyInput = document.createElement('input');
+  keyInput.type = 'hidden';
+  keyInput.name = 'comment-key';
+  keyInput.value = key;
```

Comment:
What does this part do?

Reply:
sends the key of the comment being liked as a param in the post request so that it can be found in the database. should I comment it?

Reply:
Also something to note is all the keys can be found at /list-comments which I suppose would be a security threat. This was the only way I could think of getting an identifier for the comment (`key`) comes from /list-comments but do you think I should look into a more secure way of doing this?

Reply:
Oh I see - ok so since this is just a button maybe we can simplify this and not make it a form, but instead a button. 

So createLikeButton can take in the comment as an argument. And then you can use the url to create search params to send in your post request. See the last function deleteTask() in this file: https://github.com/googleinterns/step/blob/master/walkthroughs/week-3-server/examples/todo-list/src/main/webapp/script.js

Then your AddLikeServlet would have to get that key from the request param that you sent - once you have the key, you can add the like as you're already doing. See this servlet as an example: https://github.com/googleinterns/step/blob/master/walkthroughs/week-3-server/examples/todo-list/src/main/java/com/google/sps/servlets/DeleteTaskServlet.java

Sending params like this in server requests can be super helpful. More info here https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams

Reply:
So I wrote a sendLike function like they did in the example and it works, but for some reason adding an event listener isn't working (I've also tried setting the onclick attribute).. this is in commit ab52462